code/modules/hydroponics/spreading/spreading_growth.dm

```diff
@@ -170,6 +144,58 @@
 			chem_regen_cooldown = FALSE
 
 
+/obj/effect/plant/proc/life()
+	var/turf/simulated/T = get_turf(src)
+	if(istype(T))
+		health -= seed.handle_environment(T,T.return_air(),null,1)
+	if(health < max_health)
+		//Plants can grow through closed airlocks, but more slowly, since they have to force metal to make space
+		var/obj/machinery/door/D = (locate(/obj/machinery/door) in loc)
+		if (D)
+			health += rand_between(0,0.5)
+		else
+			health += rand_between(1,2.5)
+		refresh_icon()
+		if(health > max_health)
+			health = max_health
+	else if(health == max_health && !plant && (seed.type != /datum/seed/mushroom/maintshroom))
```

Comment:
`seed.type != /datum/seed/mushroom/maintshroom`
Is it really should be handled that way?